### PR TITLE
Integrate Global Search API with Search Bar

### LINKED ISSUE #171: Integrate Global Search API with Search Bar

### PR DESCRIPTION
**Pull request by vivesing**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #171 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://globalsearchv2--icicidirect--aemsites.hlx.live/research/equity

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
NA

### CODE REVIEW

**Review by sdmcraft — COMMENTED** [review]
The search box in the test url https://globalsearchv2--icicidirect--aemsites.hlx.live/research/equity  doesn't work for me. The research api calls fail 
Is this expected?

**Comment by anagarwa** [comment]
Original Site does not perform search on single character ? its happening here https://globalsearchv2--icicidirect--aemsites.hlx.live/research/equity
is it possible to fix this ?

File: blocks/header/headerutil.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  MUTUAL_FUND: 'MF',
+  CURRENCY: 'currency',
+  COMMODITY: 'commodity',
+  KNOWLEDGE_CENTER: 'ilearn',
+  IDX: 'IDX',
+  DERIVATIVE: 'derivative',
+  BONDS: 'bonds',
+};
+
+const SITE_ROOT = 'https://www.icicidirect.com';
```
Comment (anagarwa) [inline blocks/header/headerutil.js:13]: Can we resuse constant from block-utils here ?  or if ist not there we can move this constant in block-utils

**Comment by vivesing** [comment]
> Original Site does not perform search on single character 
Okay I can add this, we can check with ICICI once if they want to allow 1 char search since the API is supporting it.

File: blocks/header/headerutil.js (hunk 1 above)
Comment (vivesing) [reply blocks/header/headerutil.js]: done moved it into block-utils.

File: blocks/header/header.js [hunk 2]
```diff
@@ -142,42 +155,41 @@ const getSearchCategoryDropDown = (fragment) => {
...
-    <div class="category equity">
+    <div class="loader hidden">
+      Searching...
+    </div>
+    <div class="category equity hidden">
       <span>Equity</span>
       <ul class="equity-list"></ul>
     </div>
-    <div class="category mf">
+    <div class="category mf hidden">
```
Comment (anagarwa) [inline blocks/header/header.js:169]: Nit: classname can be mutual-fund as complete name was used in all categories

**Review by anagarwa — APPROVED** [review]

File: blocks/header/header.js (hunk 2 above)
Comment (vivesing) [reply blocks/header/header.js:169]: have kept it intentionally to match the category keyword being passed in the API as much as possible

**Pull request merged**